oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/record-requests/utils/hooks.js

```diff
@@ -1,15 +1,42 @@
-import React, { useState } from "react";
+import React, { useState, useCallback } from "react";
+
+import _isEmpty from "lodash/isEmpty";
+import axios from "axios";
+
+import { isDeepEmpty } from "../utils";
 
 import { i18next } from "@translations/oarepo_requests_ui/i18next";
 import { Button } from "semantic-ui-react";
+import { useFormikContext } from "formik";
 
 import { REQUEST_TYPE } from "./objects";
 
 /** 
  * @typedef {import("semantic-ui-react").ConfirmProps} ConfirmProps
  */
 
-export const useConfirmDialog = (formik, sendRequest, isEventModal) => {
+export const useRequestModal = (postSubmitEvent = () => {}, onError = () => {}) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const close = useCallback(() => setIsOpen(false), []);
+  const open = useCallback(() => setIsOpen(true), []);
+
+  const onSubmit = async (asyncSubmitEvent) => {
+    try {
+      await asyncSubmitEvent();
+      close();
+      postSubmitEvent();
+    } catch (e) { 
+      onError(e);
+     }
+  };
+
+  return { isOpen, close, open, onSubmit };
+};
+
+export const useConfirmDialog = (isEventModal = false) => {
+  const { setSubmitting } = useFormikContext();
+
   /** @type {[ConfirmProps, (props: ConfirmProps) => void]} */
   const [confirmDialogProps, setConfirmDialogProps] = useState({
     open: false,
```

Comment:
everything except `open` isn't really a state, as it is static once rendered, or isn't it? open/close state is already solved in hook from oarepo-ui